src/appleseed/foundation/math/microfacet.h

```diff
@@ -331,59 +212,30 @@ class MDF
             -slope[0] * alpha_x,
             T(1.0),
             -slope[1] * alpha_y);
-        return normalize(m);
+        return normalize(sign_cos_vn * m);
     }
 
     template <typename Distribution>
-    static T do_eval_Dw(
+    T pdf_visible_normals(
         const Distribution&  mdf,
         const Vector<T, 3>&  v,
         const Vector<T, 3>&  h,
         const T              alpha_x,
-        const T              alpha_y)
+        const T              alpha_y) const APPLESEED_OVERRIDE
     {
         assert(is_normalized(v));
 
-        if (cos_theta(v) == T(0.0))
+        const T cos_theta_v = MDF<T>::cos_theta(v);
+
+        if (cos_theta_v == T(0.0))
             return T(0.0);
 
         return
-            mdf.do_eval_G1(v, h, alpha_x, alpha_y) *
+            mdf.G1(v, h, alpha_x, alpha_y) *
             std::abs(dot(v, h)) *
-            mdf.do_eval_D(h, alpha_x, alpha_y) /
-            std::abs(cos_theta(v));
+            mdf.D(h, alpha_x, alpha_y) /
+            std::abs(cos_theta_v);
```

Comment:
I would suggest moving the std::abs() statements to the same line as the mdf calls to increase readability:

    return
        mdf.G1(v, h, alpha_x, alpha_y) * std::abs(dot(v, h)) *
        mdf.D(h, alpha_x, alpha_y) / std::abs(cos_theta_v);